src/main/scala/io/qbeast/spark/delta/QbeastMetadataOperation.scala

```diff
@@ -185,7 +184,9 @@ private[delta] class QbeastMetadataOperation extends ImplicitMetadataOperation w
         errorBuilder.addSchemaMismatch(txn.metadata.schema, dataSchema, txn.metadata.id)
       }
       errorBuilder.finalizeAndThrow(spark.sessionState.conf)
-    } else txn.updateMetadata(txn.metadata.copy(configuration = configuration))
+    } else if (hasRevisionUpdate) {
```

Comment:
Let's put some comments regarding the logic we are applying to update (or not) the metadata